### ISSUE TITLE
Support Lustre 2.10.3

### ISSUE THREAD

**Issue by jgrund** [intel-hpdd/intel-manager-for-lustre#442]
Bump support to Lustre 2.10.3

**Comment by johnsonw** [comment]
**Summary of *managed mode* test for #442** using build http://jenkins.lotus.hpdd.lab.intel.com/job/manager-for-lustre/2938/arch=x86_64,distro=el7/artifact/chroma-bundles/iml-4.0.2.0.tar.gz.

*Confirmation of build*
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/1100657/34791733-db544ef2-f613-11e7-8db3-b4f22ec18754.png">

**Benign Issues**
1. Agent failed to deploy on oss1 due to the following:
```
GPG key retrieval failed: [Errno 12] Timeout on
 
http://fedora.mirrors.pair.com/epel/RPM-GPG-KEY-EPEL-7 : (28, 'Connection timed out after 30923 milliseconds')
 " [2018-01-10 12:12:24,075] bootstrap bootstrap ERROR Error from agent-bootstrap-script main(): Traceback (most recent call last): File "<stdin>", line 253, in main File "<stdin>", line 215, in install_agent File "<stdin>", line 160, in launch_command
```

I attempted to re-deploy the agent and it was successful.

2.  Clicking on the spinner to check the status of the servers being added produced the following error (I could not reproduce):
```
[root@adm ~]# journalctl --no-pager -u iml-srcmap-reverse
 -- Logs begin at Wed 2018-01-10 11:41:25 EST, end at Wed 2018-01-10 12:33:11 EST. -- Jan 10 12:23:13 adm.lfs.local systemd[1]: Started Source Map Reverse Service. Jan 10 12:23:13 adm.lfs.local systemd[1]: Starting Source Map Reverse Service... Jan 10 12:23:52
 adm.lfs.local node[21296]: "at id__in webpack:///source/iml/command/get-command-stream.js:17:6\nat webpack:///source/iml/command/command-monitor-directive.js:30:19\nat null:null:null\nat fn webpack:///node_modules/angular/angular.js:16213:0\nat fn webpack:///node_modules/angular/angular.js:26592:0\nat
 $parse webpack:///node_modules/angular/angular.js:17994:0\nat this webpack:///node_modules/angular/angular.js:18094:0\nat scope webpack:///node_modules/angular/angular.js:26597:0\nat handler webpack:///node_modules/angular/angular.js:3546:0\nat handlerWrapper
 webpack:///node_modules/angular/angular.js:3534:0”
```

```
export default (commandList: Command[]): HighlandStreamT<Command[]> => {
  const options = {
    qs: {
      id__in: commandList.map(x => x.id)
    }
  };
``

It appears that commandList was somehow null in this case.

3. After the servers finished adding and had all of their check marks the command modal showed that it was still busy and the spinner at the top right continued to spin. I tried refreshing the page and the problem was gone (no more spinner).

**Charts:**
After the filesystem was successfully created, I added a client and mounted the filesystem. I then created a large file. The charts reflected the data:

<img width="1112" alt="image" src="https://user-images.githubusercontent.com/1100657/34791451-00f2036c-f613-11e7-8cce-8fe7ceee15b3.png">

**Failover:**
After halting mds1 I received the following alerts:
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/1100657/34791562-538b3350-f613-11e7-92d2-ff72eda4e3aa.png">

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/1100657/34791629-8120c370-f613-11e7-959b-20fa43fc23e2.png">

While running on the secondary server, I logged into the client and verified that I could see the files I created before. I then brought mds1 back up and I was given the option to fail back. After failing back, all alerts went away and the filesystem was no longer running on a secondary server.

<img width="1122" alt="image" src="https://user-images.githubusercontent.com/1100657/34791716-c965e728-f613-11e7-8782-a4b854ef6197.png">

**Comment by johnsonw** [comment]
**Summary of *managed mode ZFS* test for #442** using build http://jenkins.lotus.hpdd.lab.intel.com/job/manager-for-lustre/2938/arch=x86_64,distro=el7/artifact/chroma-bundles/iml-4.0.2.0.tar.gz.

*Confirmation of build*
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/1100657/34791733-db544ef2-f613-11e7-8db3-b4f22ec18754.png">

**Charts:**
After the filesystem was successfully created, I added a client and mounted the filesystem. I then created a large file. The charts reflected the data:

<img width="1104" alt="image" src="https://user-images.githubusercontent.com/1100657/34797758-76deb02a-f627-11e7-9fd4-afdf39a48fc6.png">

**Failover:**
After halting mds1 I received the following alerts:

<img width="1120" alt="image" src="https://user-images.githubusercontent.com/1100657/34797913-f32b331a-f627-11e7-8cd2-e338ff1b27ce.png">

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/1100657/34797927-fc7884d6-f627-11e7-8a94-37c19a1f50e9.png">

While running on the secondary server, I logged into the client and verified that I could see the files I created before. I then brought mds1 back up and I was given the option to fail back. After failing back, all alerts went away and the filesystem was no longer running on a secondary server.

<img width="1116" alt="image" src="https://user-images.githubusercontent.com/1100657/34798069-9409b6f8-f628-11e7-9690-f5737dd1e378.png">

**Comment by johnsonw** [comment]
**Summary of *monitored mode ZFS* test for #442** using build http://jenkins.lotus.hpdd.lab.intel.com/job/manager-for-lustre/2938/arch=x86_64,distro=el7/artifact/chroma-bundles/iml-4.0.2.0.tar.gz.

*Confirmation of build*
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/1100657/34791733-db544ef2-f613-11e7-8db3-b4f22ec18754.png">

**Creating the filesystem:**
- After creating the zpools as discussed [here](https://intel-hpdd.github.io/Online-Help/docs/Contributor_Docs/cd_Monitored_Only_ZFS.html), I was able to detect the filesystem on the server page:

<img width="597" alt="image" src="https://user-images.githubusercontent.com/1100657/34808134-e29c8b10-f65a-11e7-8e76-eea586f70fab.png">

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/1100657/34808189-3c9f7f00-f65b-11e7-84b5-c7ba9d8817a1.png">

The filesystem was detected successfully.

**Charts:**
<img width="1098" alt="image" src="https://user-images.githubusercontent.com/1100657/34808237-8ccdc4c8-f65b-11e7-9275-cecfea1065de.png">

**Benign Error:**
- Notice in the image above that the type of filesystem is labeled as "managed." This is incorrect and should be listed as "monitored."

This scenario does not require a failover test. The filesystem was detected without error and a client was able to mount the lustre filesystem successfully. Data was generated on the charts as expected.

**Comment by jgrund** [comment]
@johnsonw Are you sure you didn't select the managed profile in the zfs MO test?

Can you try to reproduce that bug?

**Comment by johnsonw** [comment]
Pretty sure I selected monitored for all four servers but I will run the test again when I'm finished with the ldiskfs test.

**Comment by johnsonw** [comment]
**Summary of *monitored mode ldiskfs* test for #442** using build http://jenkins.lotus.hpdd.lab.intel.com/job/manager-for-lustre/2938/arch=x86_64,distro=el7/artifact/chroma-bundles/iml-4.0.2.0.tar.gz.

*Confirmation of build*
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/1100657/34791733-db544ef2-f613-11e7-8db3-b4f22ec18754.png">

**Creating the filesystem:**
- After mounting lustre on each of the servers as discussed [here](https://intel-hpdd.github.io/Online-Help/docs/Contributor_Docs/cd_Monitored_Only_ldiskfs.html), I was able to detect the filesystem on the server page:

<img width="599" alt="image" src="https://user-images.githubusercontent.com/1100657/34838913-d0cf5ba6-f6cd-11e7-9d85-bdfe4f90cf8c.png">

<img width="1091" alt="image" src="https://user-images.githubusercontent.com/1100657/34838947-efefdc72-f6cd-11e7-9170-0ce11f2e27ab.png">

**Note** Notice the filesystem type is displaying as "managed" on the dashboard even though all servers were added using the "monitored" profile.

The filesystem was detected successfully. One odd thing I noticed is that the mount command is incorrect:
<img width="807" alt="image" src="https://user-images.githubusercontent.com/1100657/34839429-94323892-f6cf-11e7-8e96-50b902056841.png">

I tried running this command on the client and it failed as I expected. I then mounted the client with: 
```
mount -t lustre 10.73.20.12@tcp0:10.73.20.11@tcp0:/fs /mnt/fs
```
I'm not sure why it's giving the wrong mount command here. Perhaps it has something to do with the way I formatted the disks.

**Charts:**
<img width="1074" alt="image" src="https://user-images.githubusercontent.com/1100657/34839350-53353f4c-f6cf-11e7-8181-c47065146ad0.png">

This scenario does not require a failover test. The filesystem was detected without error and a client was able to mount the lustre filesystem successfully. Data was generated on the charts as expected.

**Comment by johnsonw** [comment]
Pushed a fix for the filesystem label showing "managed" when it should display "monitored": https://github.com/intel-hpdd/GUI/pull/112

**Comment by johnsonw** [comment]
**Summary of *managed* upgrade from 10.0.2 to 10.0.3 test for #442** using published release of 4.0.2 and upgrading to build http://jenkins.lotus.hpdd.lab.intel.com/job/manager-for-lustre/2938/arch=x86_64,distro=el7/artifact/chroma-bundles/iml-4.0.2.0.tar.gz.

*Confirmation of build*
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/1100657/34858793-4dd5cae8-f720-11e7-9927-ff890ed115a1.png">

**Charts:**
<img width="1116" alt="image" src="https://user-images.githubusercontent.com/1100657/34858820-72d8ada6-f720-11e7-8d04-542824a0d8eb.png">

**Failover:**
I tested failover both before and after the upgrade. I received the following three alerts after halting mds1:
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/1100657/34858837-9333b5be-f720-11e7-9a51-5212c9bcb3d9.png">

<img width="1125" alt="image" src="https://user-images.githubusercontent.com/1100657/34858854-b23e5c48-f720-11e7-8bca-b372c7b4c756.png">

I then brought mds1 back online and was given the option to fail back. Failing back was successful.

<img width="573" alt="image" src="https://user-images.githubusercontent.com/1100657/34858873-d2a17592-f720-11e7-97dd-198a5caf02f9.png">

**Comment by johnsonw** [comment]
**Summary of *managed zfs* upgrade from 3.1.1.0 to 4.0.3 test for #442** using published release of 3.1.1.0 and upgrading to build http://jenkins.lotus.hpdd.lab.intel.com/job/manager-for-lustre/2938/arch=x86_64,distro=el7/artifact/chroma-bundles/iml-4.0.2.0.tar.gz.

*Confirmation of build*
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/1100657/34858793-4dd5cae8-f720-11e7-9927-ff890ed115a1.png">

**Upgrade Results:**
```
Testing YUM
Loaded plugins: fastestmirror, langpacks
Repository extras is listed more than once in the configuration
Starting Intel(R) Manager for Lustre* software upgrade
chroma-manager 3.1.1.0-4812.el7 is currently installed
Unpacking installation package el7/iml-manager-4.0.2.0.tar.gz
Upgrading Intel(R) Manager for Lustre*
/
Starting setup...

DB already accessible
Creating database tables...
Loaded 13 default power device types.
User accounts already created
API user created
Using existing (chroma configured) ntp server: localhost
Writing ntp configuration: localhost
Restarting ntp
Starting RabbitMQ...
Creating RabbitMQ user...
Creating RabbitMQ vhost...
Enabling daemons
Starting daemons
Checking service configuration...

Setup complete.
Registering profiles
chroma-manager 4.0.2.0-4999.4.gc9020d8.jenkins_manager_for_lustre_2938.el7 is currently installed

Intel(R) Manager for Lustre* software upgrade completed successfully
```

**Upgrading Agents:**
- I upgraded the agents following the document here: https://intel-hpdd.github.io/Online-Help/docs/Upgrade_Guide/Upgrade_EE-3.1-el7_to_LU-LTS-el7.html. Manually updating each agent went smoothly without any errors. The server page does show that updates are available still.

It should be noted that I never unmounted the client during the upgrade process and did not receive any errors. Throughout the process, I still had access to the files on the filesystem.

**Charts:**
<img width="1078" alt="image" src="https://user-images.githubusercontent.com/1100657/35026042-bd32e5e8-fb16-11e7-8a86-327682fb2763.png">

**Failover:**
I received the following three alerts after halting mds1:
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/1100657/35025995-628a2804-fb16-11e7-886c-df824675f157.png">

<img width="1129" alt="image" src="https://user-images.githubusercontent.com/1100657/35026000-6e19afc8-fb16-11e7-86fd-90f23cbe8786.png">

I then brought mds1 back online and was given the option to fail back. Failing back was successful.

<img width="601" alt="image" src="https://user-images.githubusercontent.com/1100657/35026164-51abf5ac-fb17-11e7-9412-6baaf7381844.png">

**Server Page**
The server page indicates that all four servers have updates ready, even after the upgrade.
<img width="1090" alt="image" src="https://user-images.githubusercontent.com/1100657/35026170-5faea7a8-fb17-11e7-82a5-2c9aa15fc11d.png">

I stopped the filesystem and ran the updates on all four servers. The update failed:
```
dnf clean all --enablerepo=*: 0
Cleaning repos: epel iml-agent fasttrack base-debuginfo epel-source
              : updates-source C7.0.1406-centosplus epel-debuginfo
              : C7.0.1406-fasttrack cr epel-testing-source c7-media
              : C7.3.1611-fasttrack C7.2.1511-extras C7.2.1511-centosplus
              : C7.1.1503-base C7.0.1406-base base-source epel-testing
              : centosplus-source C7.3.1611-centosplus lustre lustre-client
              : ngompa-dnf-el7 epel-testing-debuginfo extras-source
              : C7.1.1503-centosplus C7.1.1503-fasttrack C7.2.1511-fasttrack
              : C7.2.1511-updates base updates C7.3.1611-base
              : managerforlustre-manager-for-lustre C7.2.1511-base
              : C7.3.1611-extras C7.0.1406-updates C7.3.1611-updates
              : C7.0.1406-extras e2fsprogs extras C7.1.1503-updates
              : C7.1.1503-extras centosplus
Cleaning up Everything

Repository ngompa-dnf-el7 is listed more than once in the configuration
Repository managerforlustre-manager-for-lustre is listed more than once in the configuration
Repository epel is listed more than once in the configuration

dnf repoquery --requires --enablerepo=iml-agent chroma-agent: 0
Last metadata expiration check performed 0:00:02 ago on Tue Jan 16 23:49:18 2018.
/usr/bin/python
dnf
dnf-command(repoquery)
iml-device-scanner
iml_sos_plugin
initscripts
ntp
python(abi) = 2.7
python-argparse
python-daemon
python-requests >= 2.6.0
python-setuptools
python-tzlocal
python2-iml-common1.3
python2-tablib
python2-toolz
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
systemd-python
util-linux-ng
yum-utils

Repository ngompa-dnf-el7 is listed more than once in the configuration
Repository managerforlustre-manager-for-lustre is listed more than once in the configuration
Repository epel is listed more than once in the configuration

dnf install --allowerasing -y --exclude kernel-debug --enablerepo=iml-agent chroma-agent: 0
Last metadata expiration check performed 0:00:10 ago on Tue Jan 16 23:49:18 2018.
Dependencies resolved.
================================================================================
 Package         Arch            Version                 Repository        Size
================================================================================
Removing:
 kernel          x86_64          3.10.0-514.el7          @System          148 M

Transaction Summary
================================================================================
Remove  1 Package

Installed size: 148 M
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Erasing     : kernel-3.10.0-514.el7.x86_64                                1/1 
warning: file /lib/modules/3.10.0-514.el7.x86_64/updates: remove failed: No such file or directory
warning: file /lib/modules/3.10.0-514.el7.x86_64/extra: remove failed: No such file or directory
  Verifying   : kernel-3.10.0-514.el7.x86_64                                1/1 

Removed:
  kernel.x86_64 3.10.0-514.el7                                                  

Complete!

Repository ngompa-dnf-el7 is listed more than once in the configuration
Repository managerforlustre-manager-for-lustre is listed more than once in the configuration
Repository epel is listed more than once in the configuration
Package chroma-agent-4.0.2.0-4999.4.gc9020d8.jenkins_manager_for_lustre_2938.el7.noarch is already installed, skipping.

dnf repoquery --queryformat=%{name} %{version}-%{release}.%{arch} %{repoid} --upgrades --disablerepo=* --enablerepo=iml-agent: 0
Last metadata expiration check performed 0:01:00 ago on Tue Jan 16 23:48:34 2018.

Repository ngompa-dnf-el7 is listed more than once in the configuration
Repository managerforlustre-manager-for-lustre is listed more than once in the configuration
Repository epel is listed more than once in the configuration

grubby --default-kernel: 0
/boot/vmlinuz-3.10.0-693.11.6.el7_lustre.x86_64
```
With stacktrace:
```
Traceback (most recent call last):
  File "/usr/share/chroma-manager/chroma_core/services/job_scheduler/job_scheduler.py", line 316, in _run
    result = step.run(args)
  File "/usr/share/chroma-manager/chroma_core/models/host.py", line 1425, in run
    package.update(host, package_report)
  File "/usr/share/chroma-manager/chroma_core/models/package.py", line 85, in update
    for repo_name in repo_names.intersection(package_report):
TypeError: 'NoneType' object is not iterable
```
<img width="597" alt="image" src="https://user-images.githubusercontent.com/1100657/35026336-7fe4b4c6-fb18-11e7-96fb-14cf3e75e2ed.png">

Other than the servers not upgrading in the gui, the upgrade appears to be successful. I am able to failover and failback when a node goes offline and then comes back online. The charts continue to work as expected and I was able to keep the client connected during the upgrade.
